### PR TITLE
Disentangles blob from blob mode/removes blob mode

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
This allows blobs to get a takeover victory in any game mode, but removes the dedicated blob game mode

This means that any blob left alone in whatever mode it spawns in will eventually hit critical mass, killing everyone else on the z level and turning them to blob zombies

All blobs, including admin created/event created, spawn as an untethered overmind that can choose their spot

Because

1) I think blob mode plays out very samey (rush xrays or crew dies of attrition) and isn't much fun for either side

2) Ideally everything will one day function in a mixed mode. Blobs as a threat other people have to plan and play around rather than the sole focus that just becomes a rote numbers game will hopefully make it more entertaining

3) Need more options that function for the roundstart side antags things I will hopefully finish this week

4) If people desperately want blob mode back that'd be as easy as just spawning a couple of blobs, no ticker tracking of victory conditions etc

:cl: Kor
add: Blob is now a side antagonist.
add: Event and admin blobs will now be able to choose their spawn location.
add: Blob can now win in any mode by gaining enough tiles to reach Critical Mass.
add: Blobs that have Critical Mass have unlimited points, and a minute after achieving critical mass, they will spread to every tile on station, killing anyone still on board and ending the round.
add: Using an analyzer on a blob will now reveal its progress towards Critical Mass.
add: The blob event is now more common.
/:cl:

### CODE REVIEW

**Comment by Durkel** [comment]
Good change, fuck a dedicated blob mode.

**Comment by lzimann** [comment]
about time this happened

File: code/modules/mob/dead/observer/observer.dm [hunk 1]
```diff
@@ -304,15 +304,6 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 /mob/dead/observer/is_active()
 	return 0
 
-/mob/dead/observer/Stat()
```
Comment (lzimann) [inline code/modules/mob/dead/observer/observer.dm]: would be nice to come up with something to display blob progress to observers

Comment (KorPhaeron) [reply code/modules/mob/dead/observer/observer.dm]: I already added it to analyzers for blobs, I'll add it to examining as a ghost as well

Comment (lzimann) [reply code/modules/mob/dead/observer/observer.dm]: yea, that works

**Review by Cyberboss — COMMENTED** [review]
Buff the blob event probablility?

**Comment by KorPhaeron** [comment]
@Cyberboss Sure

I was also planning on calling a proc roundstart after antag selection but before spawning that checks config probability by mode to see if it will add a side antag, and then picking some number of extras from a weighted list (add a traitor, a ling, a nightmare, morph, slaughter demon, blob, malf, abductors, blob etc)

**Comment by KorPhaeron** [comment]
[21:11:43] Runtime in overmind.dm,67: type mismatch: cannot compare /list (/list) to 400
  proc name: Life (/mob/camera/blob/Life)

21:14:02] Runtime in theblob.dm,33: Cannot read null.blobs_legit
  usr: Blob Overmind (797) (korphaeron) (/mob/camera/blob)

Evidently I set the list up incorrectly (and blob tiles aren't adding to the list) but everything else seems to be working

File: code/game/gamemodes/blob/theblob.dm [hunk 2]
```diff
@@ -182,11 +184,7 @@
 		A.blob_act(src) //also hit everything in the turf
 
 	if(make_blob) //well, can we?
-		var/obj/structure/blob/B = new /obj/structure/blob/normal(src.loc)
-		if(controller)
-			B.overmind = controller
-		else
-			B.overmind = overmind
+		var/obj/structure/blob/B = new /obj/structure/blob/normal(src.loc, overmind)
```
Comment (ChangelingRain) [inline code/game/gamemodes/blob/theblob.dm]: should still be controller

**Comment by YakumoChen** [comment]
will blobs be buffed as a consequence because as of now, the only chance blob has in any given mode is when it spawns roundstart. Most of the way into most games science will already have xrays or other guns available. I think I've only seen a late-round blob ''''win''' maybe three times at most the entire time I have ever played ss13.

**Comment by YakumoChen** [comment]
Also blob was one of my favorite dedicated game modes, like clock cult it forces crew to band together on a single goal, even though in practice usually rounds were always the same. Also I made a lot of suggestions for improving the mode on the forums but nobody seems interested in taking them seriously

File: code/game/gamemodes/blob/theblob.dm (hunk 2 above)
Comment (ChangelingRain) [reply code/game/gamemodes/blob/theblob.dm]: well, `controller ? controller : overmind` but you get it

**Comment by KorPhaeron** [comment]
@Cyberboss 

I removed the minimum time on the blob event and bumped the probability, blobs spawning after half an hour are usually screwed anyway

**Comment by Doppeltongue** [comment]
How will blob shuttle block interact with Ratvar victory or other forced shuttle events? Clockcult has a good period of time where nobody is really on station to deal with blobs, and it would be terrible for cultists to summon Ratvar only to get killed by the blob. This could also make nuke ops way too likely to win in situations where most of the crew is dealing with blobs.

Other than that, this change might result in situations where station-based antags are forced to decide on how to work with the crew to deal with an entirely separate antag.

....Just a thought, but can blobs be affected by sigils/conversion runes/flashes?

**Comment by Supermichael777** [comment]
a disoloution of game modes probably means that only hard round enders will be cult/rev victorys and even then rev victory can keep going to deal with other threats.

**Comment by KorPhaeron** [comment]
@Doppeltongue I think antags having to balance doing their objectives vs not getting killed horribly by the biohazard would be a cool thing.

Nuke ops might have to be a side antag free mode in general though

@YakumoChen Nothing to stop people working on blob balance now, and readding it as its own game mode is as simple as spawning one roundstart

**Comment by Robustin** [comment]
Oh man, I need to take lessons from Kor on obfuscating my titles to avoid controversy.

**Comment by JJRcop** [comment]
Think you could make an antag datum that just spawns the blob?

**Comment by optimumtact** [comment]
@Robustin or you could try making changes people actually think are good.

**Comment by Robustin** [comment]
@optimumtact 

Please clarify, are non-maintainers considered people?

**Comment by lzimann** [comment]
only some of them

**Comment by Robustin** [comment]
Also please "disentangle" wizard from wizard mode next.

File: code/game/gamemodes/blob/theblob.dm (hunk 2 above)
Comment (duncathan) [reply code/game/gamemodes/blob/theblob.dm]: `(controller || overmind)` nobody remembers short circuiting for non Boolean applications

**Comment by Qbopper** [comment]
i still think robustin's prs were dumb but yeah it would be a good idea to include "and removes blob gamemode" in the title

still a good pr either way

**Comment by Qbopper** [comment]
thank

**Comment by AnonymousNow** [comment]
Oh dear. I'd be all for making Blob a true side-antag, but making it take over the station at critical mass seems a little much. I feel as though it'd be better if it could just reach a point where it spreads very quickly automatically, not instantly and round-endingly; the survivors have to flee in the five minutes it takes to creep across the station, the shuttle is called, etc.

**Pull request merged**